spec/integration/check_runner_spec.rb

```diff
@@ -0,0 +1,167 @@
+require 'webmock/rspec'
+require 'checker'
+require 'fileutils'
+
+# integration test with in-memory db, webmocked services, empty whitelist, real csvs (removed afterwards)
+# TODO: suppress logging
+RSpec.describe CheckRunner do
+  before do
+    @publishing_api_content_and_links_url = "#{Plek.new.find('publishing-api')}/v2/grouped-content-and-links"
+    @publishing_api_lookup_url = "#{Plek.new.find('publishing-api')}/lookup-by-base-path"
+    @rummager_search_url = "#{Plek.new.find('rummager')}/unified_search.json"
+
+    @csvdir = tempdir
+
+    stub_publishing_api_content
+    stub_rummager_content
+  end
+
+  after do
+    cleanup
+  end
+
+  it "runs imports, runs checks, and generates output" do
+    runner = CheckRunner.new("CHECK_OUTPUT_DIR" => @csvdir, "SUPPRESS_PROGRESS" => "y")
+
+    exit_code = runner.run
+    expect(exit_code).to eq(1)
+
+    check_csvs_are_present
+
+    check_csv_content
+  end
+
+  def check_csvs_are_present
+    expected_csvs = [
+      'BasePathsMissingFromRummager.csv',
+      'BasePathsMissingFromPublishingApi.csv',
+      'LinkedBasePathsMissingFromPublishingApi.csv',
+      'LinksMissingFromRummager.csv',
+      'LinksMissingFromPublishingApi.csv',
+      'RummagerLinksNotIndexedInRummager.csv',
+      'ExpiredWhitelistEntries.csv',
+      'RummagerRedirectedLinks.csv',
+      'RummagerRedirects.csv',
+    ]
+    actual_csvs = Dir[File.join(@csvdir, '*')].map { |f| File.basename(f) }
+    expect(actual_csvs).to contain_exactly(*expected_csvs)
+  end
+
+  def check_csv_content
+    expect(read_csv('LinksMissingFromRummager.csv')).to eq("link_type,link_content_id,content_id,publishing_app,format\norganisations,42,1,app1,format1\n")
+    expect(read_csv('BasePathsMissingFromRummager.csv')).to eq("content_id,publishing_app,format\n1,app1,format1\n")
+  end
+
+  def read_csv(csv_filename)
+    File.read(File.join(@csvdir, csv_filename))
+  end
+
+  def stub_publishing_api_content
+    # first request, starting from the nil uuid
+    stub_publishing_api_content_request(
+      request_last_seen_content_id: '00000000-0000-0000-0000-000000000000',
+      response_last_seen_content_id: '1',
+      response_results: [
+        {
+          content_id: '1',
+          content_items: [
+            {
+              locale: "en",
+              base_path: "/base_path_1",
+              publishing_app: "app1",
+              format: "format1",
+              user_facing_version: "3",
+              state: "published"
+            }],
+          links: {
+            organisations: ["42"],
+            manual: ["84"]
+          }
+        }
+      ]
+    )
+
+    # request starting from previous last_seen_content_id
+    stub_publishing_api_content_request(
+      request_last_seen_content_id: '1',
+      response_last_seen_content_id: nil,
+      response_results: []
+    )
+  end
+
+  def stub_publishing_api_content_request(request_last_seen_content_id:, response_last_seen_content_id:, response_results:)
```

Comment:
There is a central stub for this, `publishing_api_has_grouped_content_and_links` - but it's not actually merged yet. Maybe we can simplify this when we do the 2nd pass of the publishing api request. https://github.com/alphagov/gds-api-adapters/pull/510/files